### ISSUE TITLE
Steamvr not showing quest headset, but showing both controllers

### ISSUE THREAD

**Issue by cain0z** [JackD83/ALVR#149]
I have both quest and pc on v9. I have uninstalled and reinstalled it on both quest and pc.
I have tried updating the runtimes.  
Currently SteamVr is 1.10.31 i cant seem to roll back.
I have tried many different combinations
Steam with vrbeta 1.10.31 
Steam beta with vr 1.10.31
steam beta with steamvr beta 1.10.32

I have looked to see if was because managed add-ons were disabled but that didn't make a difference.

I find in the server tab
totalpackets is 0

**Comment by cain0z** [comment]
I have also tried rolling back to V8 and still the same problem.

As well as V7

I did have LIV installed, and have since removed that as well trial.

**Comment by jakefever191** [comment]
try get off the beta

**Comment by cain0z** [comment]
I have tried it while on normal steam and normal steamvr (for me 1.10.31) still no success. 
It seems to connect but not push any data through.

**Comment by JackD83** [comment]
If no packets are send/received its most likely a network/firewall issue. Please make sure the ports are open (9944 & 9945 UDP)

**Comment by cain0z** [comment]
I have opened up the ports in windows firewall and on my router

this is what the alvr status looks like 
![image](https://user-images.githubusercontent.com/52687917/77310699-7e2b5d00-6d4a-11ea-81f4-106dbd1991ac.png)

and steamvr does this 

![image](https://user-images.githubusercontent.com/52687917/77310739-8c797900-6d4a-11ea-9202-15bb809db04f.png)


I don't think i made any network changes on my router that caused it to just stop working.
anything else you would like me to look at?

**Comment by cain0z** [comment]
updated to Steam beta 1.11.3 packets are now being sent however only stuck with a white screen on the quest.

**Comment by tdbe** [comment]
I have have the exact same problem / symptoms. I can connect and start server and steam starts and plugins are on. I briefly see the controllers and the headset as not blinking, but steam asks to run the room setup before continuing. And as soon as I run room setup the hmd starts to blink and says that the headset is not visible to the sensors and I can't continue. The total packets are at 0 as well. They're both on the same network.

What kind of PC are you using? I'm trying this with a Quest, on a laptop with an old nvidia GTX860m but drivers and cuda are up to date and it's not complaining about video encoding. Using Windows 10.

Could it have something to do with whether or not Steam thinks this laptop is VR-ready?

**Comment by JackD83** [comment]
Do you use some kind of 3rd party virus scanner like Norton etc? They sometimes provide their own firewall.

But with an old nvidia GTX860m, don't bother to try. Even if you get it running it will be a slideshow

**Comment by tdbe** [comment]
> Do you use some kind of 3rd party virus scanner like Norton etc? They sometimes provide their own firewall.
> 
> But with an old nvidia GTX860m, don't bother to try. Even if you get it running it will be a slideshow

Well I did run the win firewall script. 
Also had Malwarebytes and turned it off. Also had a vpn (for the internet, not the network) but turned it off. But turning those on or off didn't make a difference. How come it can FIND the quest and CONNECT to it, but not communicate? Can you think of any way I can debug that?

A slideshow is exactly what want. ;) I'm a VR developer, and while the proper VR pc is not available, I want to quickly test my apps on -something-. I just need 10s of checking a headset every now and then as I code. I don't have graphics or care about anything. If it runs at 320x240px at 10FPS, and without oculus home, I am 10000% happy ;)

**Comment by tdbe** [comment]
I used [nmap](https://nmap.org/) to check if LAN TCP and UDP port 9944 are open. Results for both say it's open. That target IP is the IP of my Quest which I can also see on my network scan. If the quest is awake / screens are on, then the results are true (port open), otherwise closed.

`C:\WINDOWS\system32>nmap -p 9944 192.168.1.63
Starting Nmap 7.80 ( https://nmap.org ) at 2020-04-08 11:16 W. Europe Daylight Time
Nmap scan report for 192.168.1.63
Host is up (0.019s latency).

PORT     STATE SERVICE
9944/tcp open  unknown
MAC Address: 2C:26:17:56:50:8E (Oculus VR)

Nmap done: 1 IP address (1 host up) scanned in 17.97 seconds

C:\WINDOWS\system32>nmap -sU -p 9944 192.168.1.63
Starting Nmap 7.80 ( https://nmap.org ) at 2020-04-08 11:20 W. Europe Daylight Time
Nmap scan report for 192.168.1.63
Host is up (0.024s latency).

PORT     STATE         SERVICE
9944/udp open|filtered unknown
MAC Address: 2C:26:17:56:50:8E (Oculus VR)

Nmap done: 1 IP address (1 host up) scanned in 18.35 seconds

C:\WINDOWS\system32>`

**Comment by JackD83** [comment]
@tdbe if you have your vpn connected to your company, you most likely have no connection to your own network. Its a stupid security measure. 

Port 9944 is the discovery port. If thats open you can find your headset. Port 9945 is the steaming port! Both UDP

**Comment by tdbe** [comment]
Ok so while the app is open in oculus, port udp/9944 is open. And even after I start the server, only port 9944 is open. See the attached screenshot. I believe there is an error somewhere in the pipeline and not a blocked LAN port, as I did nothing but what was in the instructions, and 9944 just works, so there's no reason to think 9945 does not. Also verified that no process is using 9945 or 9943.

![alvr_01](https://user-images.githubusercontent.com/1399607/78772906-bd84bb00-7992-11ea-9e34-351cdfc7320a.PNG)

My vpn is to protect my internet traffic from my isp and anyone else. My LAN is fine and separate and not connected to my workplace or anything remote. Not using it in any scenario that makes it a stupid security measure :)

**Comment by JackD83** [comment]
Please create the folder"logs" besides "ALVR.exe" and enable debug logging in the debug tab and restart everything. You will see a log file in the logs folder.

Also the SteamVR log could be interesting

**Comment by tdbe** [comment]
> Please create the folder"logs" besides "ALVR.exe" and enable debug logging in the debug tab and restart everything. You will see a log file in the logs folder.
> 
> Also the SteamVR log could be interesting

Here's a log. I can't see anything wrong. Except I can only find "9944" in the file. At no point is it trying to connect to 9943 or 9945 or any other 994x.
There's only one error in the file, with a first encoder, but as far as I can tell it then uses another one "successfully":
```
[01:22:53.967 565] Staging Texture created
[01:22:53.967 565] Try to use VideoEncoderVCE.
[01:22:53.967 565] Initializing VideoEncoderVCE.
[01:22:53.970 567] AMF Error 1. g_AMFFactory.Init()
[01:22:53.970 567] Try to use VideoEncoderNVENC.
[01:22:53.970 567] Initializing CNvEncoder. Width=720 Height=400 Format=1 (useNV12:1)
[01:22:53.973 571] Poller::Do(). Select done. 0
[01:22:53.973 571] Try to send.
[01:22:53.973 571] Poller::Do(). Select 10000 us
[01:22:53.983 578] Poller::Do(). Select done. 0
[01:22:53.983 578] Try to send.
[01:22:53.983 578] Poller::Do(). Select 10000 us
[01:22:53.988 581] Using CUDA Device 0: GeForce GTX 860M
```

[driver.log](https://github.com/JackD83/ALVR/files/4453251/driver.log)

**Comment by JackD83** [comment]
The steamvr log can be found in C:\Program Files (x86)\Steam\logs\vrserver.txt
You can also run vrpathreg.exe in the steamvr bin folder to get all configured paths.

AlVR log looks alright.

On stupid question: You are preventing the headset from going into sleep while you try to do the room setup, right?

**Comment by tdbe** [comment]
> The steamvr log can be found in C:\Program Files (x86)\Steam\logs\vrserver.txt
> You can also run vrpathreg.exe in the steamvr bin folder to get all configured paths.
> 
> AlVR log looks alright.
> 
> On stupid question: You are preventing the headset from going into sleep while you try to do the room setup, right?

Ok here's the vrserver log. Some stuff's definitely happening or not happening here.
Also, not sure how significant this is, but on your ALVR, the hmd shows up as an icon in steam with an "H", but in the original/other ALVR repo, the hmd shows up with an "S" (but it seems to have the same behavior, same driver log also)

Yeah my finger is on the prox sensor and screens lit up the whole time.

[vrserver.txt](https://github.com/JackD83/ALVR/files/4455405/vrserver.txt)

**Comment by JackD83** [comment]
vrserver is also good. Did you setup your guardian on the Quest and are you inside the guardian while setting up SteamVR? I can only reproduce this if I'm outside of the guardian and the Quest sends all apps to standby

**Comment by tdbe** [comment]
> vrserver is also good. Did you setup your guardian on the Quest and are you inside the guardian while setting up SteamVR? I can only reproduce this if I'm outside of the guardian and the Quest sends all apps to standby

Yeah I get it, I make sure it's tracking in the main menu, I start the ALVR app, I have a postit on the prox sensor, and I keep the headset and controllers in the center of the tracking space and moving it gently while trying to connect; I don't put it face down on the table or anything. I also noticed that the app can freeze after the quest goes to sleep so I restart it every time I test.

But the way ALVR works is you make a virtual openvr headset, you connect it to steam, and this headset talks to my quest through the server client upd connection, yeah? Could it be that steamvr somehow thinks my laptop is not VR ready and so does not allow any headsets to track, or it fails internally on something related to that? Or could it be Steam Bindings related? Like it doesn't recognize the tracking binding of the headset? I only have oculus hardware on hand right now so I can't test what would happen if I plugged in a real vive in my shitty laptop. So I guess it's a mystery... :<

I did get ALVR working on a PC with a GTX970 btw. It took 2 tries but it works great (with a bit of expected hand motion delay). I really really love how I can "use" oculus hardware "on my pc" without Oculus Home or even oculus drivers <3

**Comment by JackD83** [comment]
One other thing you could try: The 860m is quite old and maybe the nvencoder is just not working at the high resolution. A similar thing can be seen with some old AMD cards and the AMF encoder. 

Try to lower the video resolution to 50% and set the video codec to h246

**Comment by tdbe** [comment]
> One other thing you could try: The 860m is quite old and maybe the nvencoder is just not working at the high resolution. A similar thing can be seen with some old AMD cards and the AMF encoder.
> 
> Try to lower the video resolution to 50% and set the video codec to h246

Hi! I also installed and tried the new version (v10) and am adding the nice new unified steamvr log.
[vrserver.txt.txt](https://github.com/JackD83/ALVR/files/4492738/vrserver.txt.txt)

Yes I set the settings to the absolute lowest possible, then restarted the ALVR application and the ALVR oculus app, and tried again and still nothing; the headset goes to untracked/standby before even the Roomscale Setup window has a chance to come up. But pressing the trigger still works always (as in, the ALVR desktop server keeps trying to open steamvr every time I press the trigger so the UDP link with the quest is maintained).

![low](https://user-images.githubusercontent.com/1399607/79565276-accdf680-80b0-11ea-81da-54edc5d8f690.PNG)
